apps/docs/.vitepress/plugins/utils/extractImports.ts

```diff
@@ -150,22 +163,28 @@ export const collectImportStatements = (lines: string[], specifiedImports: strin
        */
 
       if (matches && matches.length >= 1) {
+        // Is the current import a type import?
+        const isTypeImport = matches[1]?.trim() === 'type';
+
         // importedItems: ['Provider', 'Wallet']
         const importedItems = matches[2].replace(/[\{\}\s]/g, '').split(/\,/);
 
         // importSource: 'fuels'
         const [, importSource] = matches[3].split("'");
 
+        // source: 'type::fuels' or 'fuels'
+        const source = isTypeImport ? `${TYPE_IMPORT_PREFIX}${importSource}` : importSource;
+
         // Add collected imports 'allImportedItems' only if they were specified
         importedItems.forEach((item) => {
           if (specifiedImports.includes(item)) {
             allImportedItems.add(item);
 
-            if (!importStatements[importSource]) {
-              importStatements[importSource] = new Set();
+            if (!importStatements[source]) {
+              importStatements[source] = new Set();
             }
 
-            importStatements[importSource].add(item);
+            importStatements[source].add(item);
```

Comment:
@petertonysmith94 good stuff.

I wonder if these changes are shown just like expected on the integrated docs: https://docs.fuel.network/docs/fuels-ts/